vms/platformvm/client.go

```diff
@@ -374,6 +376,15 @@ func (c *client) GetL1Validator(
 		return L1Validator{}, 0, err
 	}
 
+	var minNonce uint64
+	if res.MinNonce != nil {
+		minNonce = uint64(*res.MinNonce)
+	}
+	var balance uint64
+	if res.Balance != nil {
+		balance = uint64(*res.Balance)
+	}
+
```

Comment:
Is there a reason this wasn't also applied to the weight? I'm assuming you did this to check vars that are meant to change/be updated

Reply:
This is to ensure they're not nil before de-referencing. `Weight` is not a pointer [here](https://github.com/ava-labs/avalanchego/pull/3843/files/5d1f16f5bf563bfacf860014896b18891cc6f439#diff-bc67b377b06e3e87764d1efb1a1dde6337ed4b902ceb0f0adba9b7257397da18R102) thus cannot be nil.

Reply:
[no action required, just curious] why is weight stored as a non-pointer?

Reply:
Weight is common in all validator types (stakers vs l1 validators), so they will always be present in json. By using a pointer we can mark it them as being "nilable" in json representations. Otherwise they will be always shown with their zero values.